src/helpers/xml-builder.js

```diff
@@ -712,49 +712,70 @@ const buildRun = async (vNode, attributes, docxDocumentInstance) => {
     const textFragment = buildTextElement(vNode.text);
     runFragment.import(textFragment);
   } else if (attributes && attributes.type === 'picture') {
-    let response = null;
 
-    const base64Uri = decodeURIComponent(vNode.properties.src);
-    if (base64Uri) {
-      response = docxDocumentInstance.createMediaFile(base64Uri);
+    const imageSource = vNode.properties.src;
+    let requiresConversion = false;
+    let isConverted = false;
+
+    if(isValidUrl(imageSource)){
+      requiresConversion = true
+      const base64String = await imageToBase64(imageSource).catch((error) => {
+        // eslint-disable-next-line no-console
+        console.warning(`skipping image download and conversion due to ${error}`);
+      });
+      if (base64String) {
+        isConverted = true;
+        vNode.properties.src = `data:${mimeTypes.lookup(imageSource)};base64, ${base64String}`;
+      }
     }
-
-    if (response) {
-      docxDocumentInstance.zip
-        .folder('word')
-        .folder('media')
-        .file(response.fileNameWithExtension, Buffer.from(response.fileContent, 'base64'), {
-          createFolders: false,
-        });
-
-      const documentRelsId = docxDocumentInstance.createDocumentRelationships(
-        docxDocumentInstance.relationshipFilename,
-        imageType,
-        `media/${response.fileNameWithExtension}`,
-        internalRelationship
-      );
-
-      const imageBuffer = Buffer.from(response.fileContent, 'base64');
-      const imageProperties = sizeOf(imageBuffer);
-
-      attributes.inlineOrAnchored = true;
-      attributes.relationshipId = documentRelsId;
-      attributes.id = response.id;
-      attributes.fileContent = response.fileContent;
-      attributes.fileNameWithExtension = response.fileNameWithExtension;
-
-      attributes.maximumWidth =
-        attributes.maximumWidth || docxDocumentInstance.availableDocumentSpace;
-      attributes.originalWidth = imageProperties.width;
-      attributes.originalHeight = imageProperties.height
-
-      computeImageDimensions(vNode, attributes);
+    
+    // we add this check because if the image is not converted, we should not proceed with the conversion
+    const shouldProceed = requiresConversion && !isConverted ? false : true;
+    if(shouldProceed){
+      let response = null;
+  
+      const base64Uri = decodeURIComponent(vNode.properties.src);
+      if (base64Uri) {
+        response = docxDocumentInstance.createMediaFile(base64Uri);
+      }
+  
+      if (response) {
+        docxDocumentInstance.zip
```

Comment:
Did you test this in scenarios where media folder is missing?

Reply:
I didn't add any image to logos or other, so I think it would be missing.
Will that be considered as scenario where we don't have media folder in main repo?

Reply:
@K-Kumar-01 yes